src/components/AutocompleteSearchBar/AutocompleteSearchBar.js

```diff
@@ -0,0 +1,150 @@
+/*
+ * This file is part of React-SearchKit.
+ * Copyright (C) 2018 CERN.
+ *
+ * React-SearchKit is free software; you can redistribute it and/or modify it
+ * under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import _ from 'lodash';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Input } from 'semantic-ui-react';
+
+class AutocompleteSearchBar extends Component {
+  constructor(props) {
+    super(props);
+    this.updateQueryString = this.props.updateQueryString;
+    this.updateSuggetions = this.props.debounce
+      ? _.debounce(this.props.updateSuggetions, this.props.debounceTime, {
+          leading: true,
+        })
+      : this.props.updateSuggetions;
+    this.clearSuggestions = this.props.clearSuggestions;
+    this.state = {
+      currentValue: this.props.queryString || '',
+    };
+    this.renderElement = props.renderElement || this._renderElement;
+    this.renderSuggestions = props.renderSuggestions || this._renderSuggestions;
+    this.handleAutocompleteChange =
+      props.handleAutocompleteChange || this._handleAutocompleteChange;
+  }
+
+  _renderSuggestions(querySuggestions) {
+    // Define actions
+    const onSuggestionSelected = async suggestion => {
+      await this.setState({
+        currentValue: suggestion,
+      });
+      this.clearSuggestions();
+      this.executeSearch();
+    };
+
+    //Render Element
+    if (querySuggestions.length === 0) {
+      return null;
+    }
+
+    return (
+      <ul>
+        {querySuggestions.map(text => (
+          <li onClick={() => onSuggestionSelected(text)} key={text}>
+            {text}
+          </li>
+        ))}
+      </ul>
+    );
+  }
+
+  _renderElement(
+    placeholder,
+    queryString,
+    querySuggestions,
+    onInputChange,
+    executeSearch
+  ) {
+    // Initialize values
+    placeholder = placeholder || 'Type something';
+    // Define actions
+    const onBtnSearchClick = (event, input) => {
+      executeSearch();
+    };
+    const onKeyPress = (event, input) => {
+      if (event.key === 'Enter') {
+        executeSearch();
+      }
+    };
+
+    //Render Element
+    return (
+      <div className="AutoCompleteText">
+        <Input
+          action={{
+            content: 'Search',
+            onClick: onBtnSearchClick,
+          }}
+          fluid
+          placeholder={placeholder}
+          onChange={(event, { value }) => {
+            onInputChange(value);
+          }}
+          value={queryString}
+          onKeyPress={onKeyPress}
+        />
+        {this.renderSuggestions(querySuggestions)}
+      </div>
+    );
+  }
+
+  _handleAutocompleteChange = () => {
+    if (this.state.currentValue.length >= 3) {
+      this.updateSuggetions(this.state.currentValue);
+    }
+  };
+
+  onInputChange = async queryString => {
+    await this.setState({
+      currentValue: queryString,
+    });
+
+    this.handleAutocompleteChange();
```

Comment:
How have you imagined that to be used from user when they override it? In general I think we have this problem that we expose props that we imply that if you override you need to execute some piece of code that was already there, e.g update component state/redux state, and this is not user friendly. I think, as I understood from what you are doing here, a prop `minCharsToAutocomplete` should be enough.

Reply:
Done. Passed the querySuggestion as parameter to ease the overload.